src/units.rs

```diff
@@ -9,7 +9,7 @@ use std::{cell::RefCell, collections::HashMap, hash::Hash as StdHash};
 
 #[derive(Debug, Clone, Copy, Eq, PartialEq, Encode, Decode, StdHash)]
 pub(crate) struct UnitCoord {
-    round: u16,
+    pub(crate) round: u16,
```

Comment:
I would strongly prefer accessors to making fields public, especially only for tests. Same for the other instances in this file.

Reply:
Good point, I'll change that.